### ISSUE TITLE
Received receipt type retry not being dealt with?

### ISSUE THREAD

**Issue by ianhook** [tgalal/yowsup#1094]
I can receive data from a certain phone number but I can not send to it.  It definitely has something to do with the axolotl RetryOutgoingReceiptProtocolEntity, but looking at the code that seems like yowsup should be sending that, not receiving it.  

```
2015-10-14 07:00:22,950 tx:
<message to="XXXX@s.whatsapp.net" type="text" id="1444806022-1">
[ENCODED DATA]
</message>

2015-10-14 07:00:23,037 rx:
<ack t="1444806022" from="XXXX@s.whatsapp.net" class="message" id="1444806022-1">
</ack>

2015-10-14 07:00:23,546 rx:
<receipt type="retry" from="XXXX@s.whatsapp.net" id="1444806022-1" t="1444806023">
<retry count="1" t="1444806022" id="1444806022-1" v="1">
</retry>
<registration>
b��
HEX:62eebbc8
</registration>
</receipt>

2015-10-14 07:00:23,547 tx:
<ack to="XXXX@s.whatsapp.net" type="retry" id="1444806022-1" class="receipt">
</ack>
```
##

**Comment by parag** [comment]
Getting same issue. Were you able to resolve this?

**Comment by Rustem** [comment]
Actually I rewrite the architecture a bit to support multiple phone numbers behind the gevent loop. I could patch asyncore loop, Networking layer and successfully tested it on 10 simultaneous numbers.

What I recomment first is to check which layers are configured in your use case. During rearchitecturing I found some bugs, for example I rewrite LoggerLayer since I could not receive messages successfully.

**Comment by hvelo** [comment]
We've got the same issue.  Checked the source code and do not have a clue to fix this.

I have server programs using Yowsup to receive and send messages.  
1. server using is sender phone number A
2. recipients using phone numbers B, C, D on mobile phones with Whatsapp official client
3. A can send messages to B, C, D without problems for weeks
4. recipient number B, changed to use another handset
5. B no longer able to receive messages from A
6. C,D can still receive messages from A

In the debug log, retry/reistration receipts are started to show up after B changed handset
```
<receipt type="retry" from="phonenumberof_B@s.whatsapp.net" id="1446454806-2" t="1446454809">
<retry count="1" t="1446454806" id="1446454806-2" v="1">
</retry>
<registration>
wJë«
HEX:774aebab
</registration>
</receipt>
```

From source code, IncomingReceiptProtocolEntity does not handle this type of receipt.    And RetryOutgoingReceiptProtocolEntity source code has an example showing the exact xml receipt, but it is supposed to send out rather than receive.
